internal/pkg/service/appsproxy/dataapps/api/rule.go

```diff
@@ -32,15 +32,20 @@ type Rule struct {
 	AuthRequired *bool         `json:"authRequired"`
 }
 
-func (r *Rule) RegisterHandler(mux *http.ServeMux, handler http.Handler) error {
+func (r *Rule) Match(req *http.Request) (bool, error) {
```

Comment:
Yes @jachym-tousek-keboola but `mux.Handle(<path>)` gives to much power to the user, and instead of preventing misuse of other options, it is better to exactly say what we support.

Reply:
We already talked a little about it in the first PR: https://github.com/keboola/keboola-as-code/pull/1701#discussion_r1565370885

Reply:
Simplicity is not touched with it:
https://github.com/keboola/keboola-as-code/blob/cfcdc177a4215665f3831bb6fdfecac22dbf6cad/internal/pkg/service/appsproxy/proxy/apphandler/apphandler.go#L122-L129

Reply:
Sorry, I should have commented it at the beginning.

Reply:
And it also allows use to pass errors from lower to upper levels, I forgot to write that.